File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()){
            return false;
        }
        HashMap<Character ,Character> map = new HashMap<>();
        for(int i =0 ; i<s.length() ; i++){
            char sc = s.charAt(i);
            char tc = t.charAt(i);
            if(!map.containsKey(sc)){
                if(!map.containsValue(tc)){
                    map.put(s.charAt(i) , t.charAt(i));
                }else{
                    return false;
                }
            }else if( map.get(sc) != tc){
                return false;
            }
        }
        return true;

        // int m1[] = new int[256];
        // int m2[] = new int[256];
        // if(s.length() != t.length()){
        //     return false;
        // }
        // for(int i=0 ; i< s.length() ; i++ ){
        //     if(m1[s.charAt(i)] != m2[t.charAt(i)]){
        //         return false;
        //     }
        //     m1[s.charAt(i)] = i+1;
        //     m2[t.charAt(i)] = i+1;
        // }
        // return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        int n1 = s.Length;
        int n2 = t.Length;
        if(n1!=n2){
            return false;
        }

        Dictionary<char , char> d = new Dictionary<char,char>();
        for(int i=0 ; i<n1 ; i++){
            if(!d.ContainsKey(s[i])){
                if(!d.ContainsValue(t[i])){
                    d.Add(s[i],t[i]);
                }else{
                    return false;
                }
                
            }else{
                if(d[s[i]] != t[i]){
                    return false;
                }
            }
        }
        return true;
    }
}